app/src/ui/dropdown/Dropdown.tsx

```diff
@@ -0,0 +1,51 @@
+import React from "react";
+import clsx from "clsx";
+import styles from "./Dropdown.module.scss";
+
+import { Grid } from "ui/grid/Grid";
+import { Container } from "react-grid-system";
+import { useToggleContext } from "hooks/useToggleContext/useToggleContext";
+import { ToggleContextController } from "context/toggle/ToggleContextController";
+import { DropdownProps, MenuProps, ItemProps, ToggleProps } from "./Dropdown.types";
+
+export const Dropdown: React.FC<DropdownProps> & {
+  Toogle: React.FC<ToggleProps>;
+  Menu: React.FC<MenuProps>;
+  Item: React.FC<ItemProps>;
+} = ({ children, className }) => {
+  return (
+    <div className={clsx(styles["dropdown"], className)}>
+      <ToggleContextController>{children}</ToggleContextController>
+    </div>
+  );
```

Comment:
Normally, you'll put context controllers on the parent layouts. 
Since this will be used everywhere, let's put it either in `AppLayout` or `AuthLayout`. 

Let's make sure that triggering the `isVisible` toggle only displays the current component and not others, I'm not sure about this, but it may activate all `isVisible` references. 

If this happens, then probably `ToggleContext` is better here, but like this: 

```suggestion
  return (
    <ToggleContextController>
       <div className={clsx(styles["dropdown"], className)}>
         {children}
      </div>
    </ToggleContextController>
  );
```

Reply:
@Ktoxcon I'm thinking that `ToggleContextController` is too much for such a simple `useState`. Normally you'd use a context hook to keep state across multiple pages and components, but since this is relative to each component that needs to be toggled, I think the right way is to create a simple: `useToggle` hook, with just the 

```
const [display, setDisplay] = useState(false);

return { isVisible, setIsVisible }
```

And that's it.